catalog/db_scanner.go

```diff
@@ -0,0 +1,28 @@
+package catalog
+
+type DBScannerOptions struct {
+	BufferSize       int
+	FilterDeleted    bool
+	After            string
+	AdditionalFields []string
+}
+
+type DBScanner interface {
+	Next() bool
+	Value() *DBScannerEntry
+	Err() error
+}
+
+func ScanDBEntryUntil(s DBScanner, ent *DBScannerEntry, p string) (*DBScannerEntry, error) {
+	for ent == nil || ent.Path < p {
+		_ = s.Next()
```

Comment:
This scanner seems to return end by `nil` value, rather than by returning `false` from `Scanner.Next`.  Isn't this a change from regular iterator semantics?

Reply:
Next() returns boolean if we read the next value  - false on end-of-file or err. I abused the fact that at the end we return nil as Value().  but I can do better...